frameworks/FEDOT/setup.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+HERE=$(dirname "$0")
+VERSION=${1:-"stable"}
+REPO=${2:-"https://github.com/aimclub/FEDOT.git"}
+PKG=${3:-"fedot"}
+if [[ "$VERSION" == "latest" ]]; then
+    VERSION="master"
+fi
+
+# creating local venv
+. ${HERE}/../shared/setup.sh ${HERE} true
+
+RAWREPO=$(echo ${REPO} | sed "s/github\.com/raw\.githubusercontent\.com/")
+if [[ "$VERSION" == "stable" ]]; then
+    PIP install --no-cache-dir -U ${PKG}
+elif [[ "$VERSION" =~ ^[0-9] ]]; then
+    PIP install --no-cache-dir -U ${PKG}==${VERSION}
+else
+    TARGET_DIR="${HERE}/lib/${PKG}"
+    rm -Rf ${TARGET_DIR}
+    git clone --depth 1 --single-branch --branch ${VERSION} --recurse-submodules ${REPO} ${TARGET_DIR}
+    PIP install -U -e ${HERE}/lib/${PKG}
+fi
```

Comment:
The setup file should write the installed version to a `.setup` file, see e.g., the autosklearn integration script:
https://github.com/openml/automlbenchmark/blob/12046acc4824dd48414c0543b518cd628490a12d/frameworks/autosklearn/setup.sh#L44-L49

This makes it possible to detect what version is already installed so installation can be skipped on subsequent runs.

Reply:
I extened the script with the proposed part: [link](https://github.com/openml/automlbenchmark/pull/563/files#diff-6aa5bff3d1573801f5d2a067bbc8acacd6eedef2a9b3f6355a97853d25e77db5R25-R30).